consul-lambda-registrator/structs/service.go

```diff
@@ -0,0 +1,131 @@
+package structs
+
+import (
+	"fmt"
+	"net/url"
+	"strconv"
+	"strings"
+)
+
+const (
+	internal        = "internal"
+	version         = "v1"
+	internalVersion = internal + "-" + version
+)
+
+type Service struct {
+	Name        string
+	Port        int
+	Namespace   string
+	Partition   string
+	Datacenter  string
+	TrustDomain string
+	Subset      string
+}
+
+// ParseService parses a string in unlabeled upstream format into a Service instance.
+func ParseService(s string) (Service, error) {
```

Comment:
Maybe something like `ParseUpstream` or `UpstreamToService` would be more descriptive?

Reply:
I like `ParseUpstream` since we're parsing a formatted string. I actually had it named `ParseUpstream` at one point.. Don't remember why I changed it..